src/wallet/rpc/backup.cpp

```diff
@@ -246,7 +249,17 @@ RPCHelpMan importaddress()
     std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
     if (!pwallet) return UniValue::VNULL;
 
-    EnsureLegacyScriptPubKeyMan(*pwallet, true);
+    // Use legacy spkm only if the wallet does not support descriptors.
+    bool use_legacy = !pwallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS);
+    if (!use_legacy) {
+        // As we don't allow mixing watch-only descriptors with spendable ones,
+        // ensure legacy spkm if wallet is blank and descriptors are not supported.
+        // (blank check is done inside 'EnsureLegacyScriptPubKeyMan()')
+        if (!pwallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
+            EnsureLegacyScriptPubKeyMan(*pwallet, /*also_create=*/true);
```

Comment:
I'm confused: `!use_legacy` means we have a descriptor wallet. We should never create a `LegacyScriptPubKeyMan` in that case. I don't understand what this `if` block is trying to achieve, but I think it should `throw JSONRPCError(RPC_WALLET_ERROR, "Use 'importdescriptors' instead"`

Reply:
yeah ok. No legacy spkm can be created here because `EnsureLegacyScriptPubKeyMan` internally calls to `SetupLegacyScriptPubKeyMan` which returns early if the `WALLET_FLAG_DESCRIPTORS` is set.

I thought that we had a release version that upgraded the legacy wallet into a descriptors wallet, letting the user import descriptors while the legacy spkm is available. But.. seems that we never allowed it, at least not officially.

Good, thanks.

Reply:
> upgraded the legacy wallet into a descriptors wallet

I think it's best to have only one way to go from a legacy wallet to a descriptor wallet (the migration RPC), so in the long run it's easier to keep testing this stuff.

Reply:
> > upgraded the legacy wallet into a descriptors wallet
> 
> I think it's best to have only one way to go from a legacy wallet to a descriptor wallet (the migration RPC), so in the long run it's easier to keep testing this stuff.

yeah, agree.